pkg/providers/version/version.go

```diff
@@ -0,0 +1,61 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/patrickmn/go-cache"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/logging"
+
+	"github.com/aws/karpenter-core/pkg/utils/pretty"
+)
+
+const (
+	kubernetesVersionCacheKey = "kubernetesVersion"
+)
+
+type Provider struct {
+	cache               *cache.Cache
+	cm                  *pretty.ChangeMonitor
+	kubernetesInterface kubernetes.Interface
+}
+
+func NewProvider(kubernetesInterface kubernetes.Interface, cache *cache.Cache) *Provider {
+	return &Provider{
+		cm:                  pretty.NewChangeMonitor(),
+		cache:               cache,
+		kubernetesInterface: kubernetesInterface,
+	}
+}
+
+func (p *Provider) KubeServerVersion(ctx context.Context) (string, error) {
```

Comment:
```suggestion
func (p *Provider) Get(ctx context.Context) (string, error) {
```

Maybe just `Get` since it's assumed that this version provider is providing the kubernetes server version